### PR TITLE
Feature/network

### PR DESCRIPTION
**Pull request by Menighin**
Hi again Alex!

Here is the pull request with the Network component.
In this pull request I:

- Created the Network Component
- Created an example for using and testing the network
- Updated the building to build the network component

Feel free to suggest changes or anything,

Regards,

### CODE REVIEW

File: src/components/Timeline.vue [hunk 1]
```diff
@@ -1,185 +1,184 @@
...
-  'groupDragged',
-  'changed',
-  'rangechange',
-  'rangechanged',
-  'select',
-  'itemover',
-  'itemout',
-  'timechange',
-  'timechanged',
+    'click',
```
Comment (alexcode) [inline src/components/Timeline.vue]: As a whole it looks good but is the indentation align with eslint airbnb-base rules ?

File: src/components/Network.vue [hunk 2]
```diff
@@ -117,11 +117,11 @@ export default {
...
             this.network.moveTo(event, callback);
         },
-        canvasToDom(x, y) {
-            return this.network.canvasToDom(x, y);
+        canvasToDom(p) {
+            return this.network.canvasToDOM(p);
         },
-        DomToCanvas(x, y) {
-            return this.network.DomToCanvas(x, y);
+        DomToCanvas(p) {
```
Comment (alexcode) [inline src/components/Network.vue]: Shouldn't it be either lowerCamelCase `domToCanvas ` either consistent with visjs API `DOMToCanvas` ?

File: src/App.vue [hunk 3]
```diff
@@ -1,259 +1,269 @@
...
+		<hr>
+
+		<h2>Graph2d</h2>
+		<graph-2d ref="graph2d" :items="graph2d.items" :groups="graph2d.groups" :options="graph2d.options"></graph-2d>
+
+	</div>
 </template>
 
 <script>
+import Timeline from "./components/Timeline.vue";
```
Comment (alexcode) [inline src/App.vue]: Throughout the package, JS code is preferably single quoted as per https://github.com/airbnb/javascript#strings--quotes. We shall keep a consistency here as well in order to gain readability.

File: src/main.build.js [hunk 4]
```diff
@@ -3,7 +3,7 @@ import Graph2d from './components/Graph2d.vue';
...
 
 export {
-    Timeline,
-    Graph2d,
-    Network
-};
+	Timeline,
+	Graph2d,
+	Network
+};
```
Comment (alexcode) [inline src/main.build.js]: Please fix the no new line error

File: src/main.build.js [hunk 5]
```diff
@@ -3,7 +3,7 @@ import Graph2d from './components/Graph2d.vue';
 import Network from './components/Network.vue';
 
 export {
-    Timeline,
-    Graph2d,
-    Network
-};
+	Timeline,
```
Comment (alexcode) [inline src/main.build.js]: Indent fix as per https://github.com/airbnb/javascript#whitespace--spaces

File: src/components/Network.vue (hunk 2 above)
Comment (Menighin) [reply src/components/Network.vue]: Yep, you are right. Changed it to `domToCanvas`.

File: src/App.vue (hunk 3 above)
Comment (Menighin) [reply src/App.vue]: My bad. I used auto formatting in a hurry and didnt notice it changed all single quotes to double. I pushed the fix.

File: src/main.build.js (hunk 4 above)
Comment (Menighin) [reply src/main.build.js]: This one I didn't get... What you mean by "no new line error"?

**Comment by Menighin** [comment]
Im sorry about all the mess with the indentations n_n'
I didnt touch any of your code for Timeline or Graph2d though. Only the indentation and quotes were automatically changed.

File: src/main.build.js (hunk 4 above)
Comment (alexcode) [reply src/main.build.js]: It's just poping in the git review. You should just add a new line at the end of the js file.
Here is why https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline

**Comment by alexcode** [comment]
Don't worry, is just that I want the package to be with the same style as it helps readability and contribution. Again, thanks a lot for your work.

**Comment by Menighin** [comment]
Now I think I got everything right Alex. All indentations with 2 whitespaces, new line at the end of .js files. Did I miss something?

**Comment by alexcode** [comment]
Thank you @Menighin ;)

**Pull request merged**